### PR TITLE
feat(next): ssr support

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add "withLogtoSsr" to support [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props)

## Testing

Local tested with sample.

### CODE REVIEW

File: packages/next-sample/pages/profile-ssr.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    res.setHeader('location', '/api/login');
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    res.statusCode = 302;
+    res.end();
+  }
+
+  return {
+    props: { user },
+  };
+});
```
Comment (demonzoo) [inline packages/next-sample/pages/profile-ssr.tsx:64]: 1. Maybe we should put this on top of the file?
2. Seems we don't need to `export`

**Review by demonzoo — APPROVED** [review]

File: packages/next-sample/pages/profile-ssr.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/next-sample/pages/profile-ssr.tsx:64]: export this function as a requirement of Next.js, https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props

Comment (wangsijie) [reply packages/next-sample/pages/profile-ssr.tsx:64]: Also, put this after "default export(component)" is common.

File: packages/next/src/index.ts [hunk 2]
```diff
@@ -97,14 +105,27 @@ export default class LogtoClient {
...
+    return {
       cookieName: `logto:${this.config.appId}`,
       password: this.config.cookieSecret,
       cookieOptions: {
         secure: this.config.cookieSecure,
         maxAge: 14 * 24 * 60 * 60,
       },
-    });
+    };
+  }
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/next/src/index.ts]: Of couse, updated.

**Pull request merged**